Public/Src/Engine/Processes/ExternalToolSandboxedProcess.cs

```diff
@@ -0,0 +1,239 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.ContractsLight;
+using System.Text;
+using System.Threading.Tasks;
+using BuildXL.Interop;
+using BuildXL.Native.IO;
+using BuildXL.Utilities.Tasks;
+
+namespace BuildXL.Processes
+{
+    /// <summary>
+    /// Sanboxed process that will be executed by an external tool.
+    /// </summary>
+    public class ExternalToolSandboxedProcess : ExternalSandboxedProcess
+    {
+        /// <summary>
+        /// Relative path to the default tool.
+        /// </summary>
+        public const string DefaultToolRelativePath = @"tools\SandboxedProcessExecutor\SandboxedProcessExecutor.exe";
+

```

Comment:
If you do this you must guard usage of the feature (on the commandline parser to be windows only and fail for the Mac)